File: gamepack/Decompiled/AttackOption.java
public class AttackOption implements Enumerated {
   static final AttackOption ac = new AttackOption(2);
   static final AttackOption af = new AttackOption(-1);
   static final AttackOption an = new AttackOption(0);
   static final AttackOption aw = new AttackOption(1);
   final int au;

   AttackOption(int var1) {
      this.au = 521487647 * var1;
   }

   public int ordinal() {
      return this.au * -1923877153;
   }
}

File: find-names/src/main/java/mapper/identifiers/classes/AttackOption.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.annotations.DependsOn
import mapper.annotations.MethodParameters
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.wrappers.ClassWrapper
import mapper.wrappers.FieldWrapper
import mapper.wrappers.MethodWrapper
import org.objectweb.asm.Type.INT_TYPE

@DependsOn(Enumerated::class)
class AttackOption : IdentityMapper.Class() {

    override val predicate = predicateOf<ClassWrapper> { it.instanceFields.size == 1 }
            .and { it.interfaces.contains(type<Enumerated>()) }
            .and { it.instanceMethods.size == 1 }
            .and { it.instanceFields.all { it.type == INT_TYPE } }
            .and { it.staticFields.count { it.type == it.klass.type } == 4 }

    @MethodParameters()
    @DependsOn(Enumerated.ordinal::class)
    class ordinal : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.mark == method<Enumerated.ordinal>().mark }
    }

    class id : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == INT_TYPE }
    }
}